pinot-segment-local/src/main/java/org/apache/pinot/segment/local/segment/creator/impl/SegmentColumnarIndexCreator.java

```diff
@@ -585,6 +585,7 @@ public static void addColumnMetadataInfo(PropertiesConfiguration properties, Str
         String.valueOf(columnIndexCreationInfo.getTotalNumberOfEntries()));
     properties.setProperty(getKeyFor(column, IS_AUTO_GENERATED),
         String.valueOf(columnIndexCreationInfo.isAutoGenerated()));
+    properties.setProperty(getKeyFor(column, SCHEMA_MAX_LENGTH), String.valueOf(fieldSpec.getMaxLength()));
```

Comment:
We don't need to store this property for all fields. I'd suggest revisiting this PR after wrapping up #13103

Reply:
Agreed! During persistence, we can optimise the logic a bit to only include STRING fields with certain MaxLengthStrategies. Will update post #13103

Reply:
hey @Jackie-Jiang updated this PR based on #13103 changes.